src/main/kotlin/ink/abb/pogo/scraper/controllers/BotController.kt

```diff
@@ -56,4 +69,202 @@ class BotController {
+        return pokemon!!.renamePokemon(newName).toString()
+    }
+
+    @RequestMapping(value = "/bot/{name}/pokemon/{id}/candies", method = arrayOf(RequestMethod.GET))
+    fun getPokemonCandies(
+            @PathVariable name: String,
+            @PathVariable id: Long
+    ): String {
+        val pokemon: Pokemon? = service.getBotContext(name).api.inventories.pokebank.pokemons.find { it.id == id }
+        return ""+pokemon!!.candy
+    }
+
+    @RequestMapping("/bot/{name}/items")
+    fun listItems(@PathVariable name: String): List<ItemData> {
+
+        val data = service.getBotContext(name).api.inventories.itemBag.items
+        val returnData = mutableListOf<ItemData>()
+
+        for (item in data) {
+            returnData.add(ItemData().buildFromItem(item))
+        }
+
+        return returnData
+    }
+
+    @RequestMapping("/bot/{name}/item/{id}/drop/{quantity}")
+    fun dropItem(
+            @PathVariable name: String,
+            @PathVariable id: Int,
+            @PathVariable quantity: Int
+    ): String {
+
+        val itemBag: ItemBag = service.getBotContext(name).api.inventories.itemBag
+        val item: Item? = itemBag.items.find { it.itemId.number == id }
+
+        if (quantity > item!!.count) {
+            return "Not enough items to drop " + item.count
+        } else {
+            Log.magenta("REST API : dropping " + quantity + " " + item.itemId.name)
+            return itemBag.removeItem(item.itemId, quantity).toString()
+        }
+    }
+
+    @RequestMapping("/bot/{name}/useIncense")
+    fun useIncense(@PathVariable name: String): String {
+        val itemBag = service.getBotContext(name).api.inventories.itemBag
+        val count = itemBag.items.find { it.itemId == ItemIdOuterClass.ItemId.ITEM_INCENSE_ORDINARY }?.count
+
+        if (count == 0) {
+            return "Not enough incense"
+        } else {
+            itemBag.useIncense()
+            return "SUCCESS"
+        }
+    }
+
+    @RequestMapping("/bot/{name}/useLuckyEgg")
+    fun useLuckyEgg(@PathVariable name: String): String {
+        val itemBag = service.getBotContext(name).api.inventories.itemBag
+        val count = itemBag.items.find { it.itemId == ItemIdOuterClass.ItemId.ITEM_LUCKY_EGG }?.count
+
+        if (count == 0) {
+            return "Not enough lucky egg"
+        } else {
+            return itemBag.useLuckyEgg().toString()
+        }
+    }
+
+    @RequestMapping("/bot/{name}/location/{latitude}/{longitude}")
+    fun changeLocation(
+            @PathVariable name: String,
+            @PathVariable latitude: Double,
+            @PathVariable longitude: Double
+    ): String {
+        val ctx: Context = service.getBotContext(name)
+
+        // Stop walking
+        ctx.pauseWalking.set(true)
+
+        ctx.lat.set(latitude)
+        ctx.lng.set(longitude)
+
+        ctx.pauseWalking.set(false)
+
+        ctx.server.setLocation(latitude, longitude)
```

Comment:
When the bot is currently walking this will immediately reset your location back to the next value on the path.